cpp/src/arrow/util/time.h

```diff
@@ -34,16 +34,8 @@ enum DivideOrMultiply {
 };
 
 // TimestampType -> TimestampType
-static std::pair<DivideOrMultiply, int64_t> kTimestampConversionTable[4][4] = {
-    // TimestampType::SECOND
-    {{MULTIPLY, 1}, {MULTIPLY, 1000}, {MULTIPLY, 1000000}, {MULTIPLY, 1000000000}},
-    // TimestampType::MILLI
-    {{DIVIDE, 1000}, {MULTIPLY, 1}, {MULTIPLY, 1000}, {MULTIPLY, 1000000}},
-    // TimestampType::MICRO
-    {{DIVIDE, 1000000}, {DIVIDE, 1000}, {MULTIPLY, 1}, {MULTIPLY, 1000}},
-    // TimestampType::NANO
-    {{DIVIDE, 1000000000}, {DIVIDE, 1000000}, {DIVIDE, 1000}, {MULTIPLY, 1}},
-};
+ARROW_EXPORT extern const std::pair<DivideOrMultiply, int64_t>
+    kTimestampConversionTable[4][4];
```

Comment:
The problem here is that the compiler won't be able to optimize divide-by-constant, which is very slow by default.

Reply:
(by "constant" I mean known compile-time constant)

Reply:
I see, I wasn't aware that accessing this matrix happened on a hot path. I'll see if making it constexpr fixes the warning otherwise I'll let it go